framework/base/Module.php

```diff
@@ -571,32 +571,43 @@ public function createController($route)
             return false;
         }
 
-        if (strpos($route, '/') !== false) {
-            list($id, $route) = explode('/', $route, 2);
-        } else {
-            $id = $route;
-            $route = '';
+        // module and controller map take precedence
+        $routeInfo=pathinfo($route);
+        $id =$routeInfo['dirname']!=='.'?$routeInfo['dirname']:$routeInfo['basename'];
+        $route = $routeInfo['dirname']!=='.'?$routeInfo['basename']:'';
+        if (isset($this->controllerMap[$id.'/'.$route])) {
+            $controllerClass= $this->controllerMap[$id.'/'.$route];
+            $id=$id.'/'.$route;
+            $route='';
+            
+        }elseif (isset($this->controllerMap[$id])) {
+            $controllerClass= $this->controllerMap[$id];
         }
 
-        // module and controller map take precedence
-        if (isset($this->controllerMap[$id])) {
-            $controller = Yii::createObject($this->controllerMap[$id], [$id, $this]);
+        if (isset($controllerClass)) {
+            $controller = Yii::createObject($controllerClass, [$id, $this]);
             return [$controller, $route];
         }
-        $module = $this->getModule($id);
-        if ($module !== null) {
-            return $module->createController($route);
+
+        if (strpos($id, '/') !== false) {
+            list($moduleId, $moduleRoute) = explode('/', $id, 2);
+            $moduleRoute.=$route!==''?'/'.$route:'';
+        } else {
+            $moduleId = $id;
+            $moduleRoute = $route;
         }
 
-        if (($pos = strrpos($route, '/')) !== false) {
-            $id .= '/' . substr($route, 0, $pos);
-            $route = substr($route, $pos + 1);
+        $module = $this->getModule($moduleId);
+        if ($module !== null) {
+            return $module->createController($moduleRoute);
         }
 
-        $controller = $this->createControllerByID($id);
-        if ($controller === null && $route !== '') {
-            $controller = $this->createControllerByID($id . '/' . $route);
-            $route = '';
+        $controller = $this->createControllerByID($id . '/' . $route);
+        if ($controller !== null ) {
+            $id=$id.'/'.$route;
```

Comment:
New id isn't used anywhere.

Reply:
Yeah sure it was not necessary necessary to initialize it. but at that step the controller ID is that` $id.'/'.$route` string and `actionId` ($route) is an empty string